public/Students/Leo Jayachandran/DynamicIndex/dynamicIndexMaker.js

```diff
@@ -144,10 +146,36 @@
 				cutString = cutString.slice(termEnd + termEndString.length);
 				newTerm.pages.push(newPage);
 				
-			}
-			pageList.taggedTerms.push(newTerm);
+            }
+            
 			//update termcutstring
-			termCutString = termCutString.slice(postPagelistPosition + postPagelist.length);
+            termCutString = termCutString.slice(postPagelistPosition + postPagelist.length);
+            
+            var canPush = true;
+            //Filter term
+            
+            for (var e = 0; e < indexRequirements.length; e++) { // Inclusions
+                if (canPush && newTerm.name.includes(indexRequirements[e])) { // if requirement is present, trim and continue check, if term is absent, 
+                    if (indexRequirements[e].endsWith("@")) { // Trim the term name if filter ends with @
+                        newTerm.name = newTerm.name.replace(indexRequirements[e], "").trim();
+                    }
+                } else { // stop checking
+                    canPush = false;
+                    break;
+                }
+            }
+            
+            for (var e = 0 ; e < indexExclusions.length; e++) { // Exclusions
+                if (canPush && newTerm.name.toLowerCase().includes(indexExclusions[e])) { //Stop check if fails
+                    canPush = false;
+                    break;
+                }
+            }
+            //Push new term
+            if (canPush) {
+                pageList.taggedTerms.push(newTerm);
+            }
+            
 		}
 		return pageList;
```

Comment:
If you are using some sort of IDE, would you be able to Beautify the code so that the indentation is correct? I assume that this curly brace is matched correctly but I can't visually tell due to the weird indentation.